src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/SetAzureSqlServerDnsAlias.cs

```diff
@@ -0,0 +1,129 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Common.Authentication.Models;
+using Microsoft.Azure.Commands.Sql.Common;
+using Microsoft.Azure.Commands.Sql.ServerDnsAlias.Services;
+using Microsoft.Azure.Commands.Sql.ServerDnsAlias.Model;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System;
+
+namespace Microsoft.Azure.Commands.Sql.ServerDnsAlias.Cmdlet
+{
+	/// <summary>
+	/// Defines the Set-AzureRmSqlServerDnsAlias cmdlet
+	/// </summary>
+	[Cmdlet(VerbsCommon.Set, "AzureRmSqlServerDnsAlias",
+		SupportsShouldProcess = true,
+ConfirmImpact = ConfirmImpact.Medium)]
+	[OutputType(typeof(Model.AzureSqlServerDnsAliasModel))]
+	public class SetAzureSqlServerDnsAlias : AzureSqlCmdletBase<IEnumerable<AzureSqlServerDnsAliasModel>, AzureSqlServerDnsAliasAdapter>
+	{
+		/// <summary>
+		/// Template to generate the Server Dns Alias Id
+		/// </summary>
+		public static string ServerDnsAliasIdTemplate = "/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Sql/servers/{2}/dnsAliases/{3}";
+
+		/// <summary>
+		/// Gets or sets the name of the Azure Sql Server Dns Alias
+		/// </summary>
+		[Parameter(Mandatory = true,
+			HelpMessage = "The Azure Sql Server Dns Alias name.")]
+		[Alias("Name")]
+		[ValidateNotNullOrEmpty]
+		public string DnsAliasName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the new server to which alias should point
+		/// </summary>
+		[Parameter(Mandatory = true,
+			HelpMessage = "The name of Azure Sql Server to which alias should point.")]
+		[ValidateNotNullOrEmpty]
+		public string TargetServerName { get; set; }
```

Comment:
shouldn't there be a TargetServerResourceGroupName param? and maybe TargetServerSubscriptionId?